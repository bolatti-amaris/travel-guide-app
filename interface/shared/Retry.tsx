import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

interface RetryProps {
  message: string;
  onRetry: () => void;
}

const Retry = ({ message, onRetry }: RetryProps) => {
  return (
    <View
      testID="retry-wrapper"
      className="flex-1 justify-center items-center bg-background dark:bg-background-dark px-4"
    >
      <Text
        testID="retry-message"
        className="text-base text-text dark:text-text-dark text-center mb-4 font-work-light"
      >
        {message}
      </Text>

      <TouchableOpacity
        testID="retry-button"
        onPress={onRetry}
        className="bg-tint dark:bg-tint-dark px-5 py-2 rounded-sm"
      >
        <Text
          testID="retry-button-label"
          className="text-background dark:text-background-dark font-work-medium"
        >
          Retry
        </Text>
      </TouchableOpacity>
    </View>
  );
};

export default Retry;
