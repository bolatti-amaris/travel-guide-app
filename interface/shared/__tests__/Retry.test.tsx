import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import Retry from '../Retry';

describe('Retry', () => {
  it('renders message and button correctly', () => {
    const { getByTestId } = render(
      <Retry message="Something went wrong." onRetry={() => {}} />,
    );

    expect(getByTestId('retry-wrapper')).toBeTruthy();
    expect(getByTestId('retry-message').props.children).toBe(
      'Something went wrong.',
    );
    expect(getByTestId('retry-button')).toBeTruthy();
    expect(getByTestId('retry-button-label').props.children).toBe('Retry');
  });

  it('applies correct classNames to wrapper and button', () => {
    const { getByTestId } = render(
      <Retry message="Try again!" onRetry={() => {}} />,
    );

    expect(getByTestId('retry-wrapper').props.className).toContain('flex-1');
    expect(getByTestId('retry-message').props.className).toContain(
      'text-center',
    );
    expect(getByTestId('retry-button-label').props.className).toContain(
      'text-background',
    );
  });

  it('calls onRetry when button is pressed', () => {
    const mockRetry = jest.fn();
    const { getByTestId } = render(
      <Retry message="Try again!" onRetry={mockRetry} />,
    );

    fireEvent.press(getByTestId('retry-button'));
    expect(mockRetry).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const tree = render(
      <Retry message="Please try again" onRetry={() => {}} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
