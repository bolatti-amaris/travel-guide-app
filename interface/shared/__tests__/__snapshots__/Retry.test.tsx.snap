// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Retry matches snapshot 1`] = `
<View
  className="flex-1 justify-center items-center bg-background dark:bg-background-dark px-4"
  testID="retry-wrapper"
>
  <Text
    className="text-base text-text dark:text-text-dark text-center mb-4 font-work-light"
    testID="retry-message"
  >
    Please try again
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="retry-button"
  >
    <Text
      className="text-background dark:text-background-dark font-work-medium"
      testID="retry-button-label"
    >
      Retry
    </Text>
  </View>
</View>
`;
